code/modules/manufactorio/machines/sorter.dm

```diff
@@ -0,0 +1,149 @@
+/obj/machinery/power/manufacturing/sorter
+	icon_state = "router"
+	name = "conveyor sort-router"
+	desc = "Pushes things on it to its sides following set criteria, set via multitool."
+	layer = BELOW_OPEN_DOOR_LAYER
+	density = FALSE
+	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND
+	circuit = /obj/item/circuitboard/machine/manurouter
+	/// for mappers; filter path = list(direction, value), otherwise a list of initialized filters
+	var/list/sort_filters = list()
+	/// dir to push to if there is no criteria
+	var/dir_if_not_met
+	/// timer id of the thing that makes stuff move
+	var/delay_timerid
+	/// max filters
+	var/max_filters = 10
+
+/obj/machinery/power/manufacturing/sorter/Initialize(mapload)
+	. = ..()
+	if(isnull(dir_if_not_met))
+		dir_if_not_met = dir
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+	for(var/i=1, i <= length(sort_filters), i++)
```

Comment:
```suggestion
	for(var/i in 1 to length(sort_filters))
```
The operations inside the loop don't seem to affect the length of sort_filters or do anything fancy on the loop itself, so there's no need to explicify the do statement I guess.